addons/stock/models/stock_move.py

```diff
@@ -545,6 +537,63 @@ def _delay_alert_log_activity(self, mode, move_orig):
                 user_id=doc.user_id.id or SUPERUSER_ID
             )
 
+    def _delay_alert_check(self, new_date):
+        """Set an alert on late moves by using the `delay_alert_date` field.
+        The alert is always on the move that cannot be done because its preceding moves are late.
+        """
+        self.ensure_one()
+        if self.state in ('done', 'cancel'):
+            return
+
+        ontime_moves = self.browse()
+        late_moves = self.browse()
+
+        # Check if `self` is scheduled after the next moves. If so, the next moves are late.
+        next_done_moves = self.browse()
+        next_nondone_moves = self.browse()
+        next_moves_dates = []
+        for move in self.move_dest_ids:
+            if move.state == 'done':
+                next_done_moves |= move
+            elif move.state != 'cancel':
+                next_nondone_moves |= move
+        next_moves_dates += next_done_moves.mapped('date')
+        next_moves_dates += next_nondone_moves.mapped('date_expected')
+        if next_moves_dates:
+            next_moves_date = min(next_moves_dates)
+            if new_date > next_moves_date:
+                late_moves |= next_nondone_moves
+            else:
+                ontime_moves |= next_nondone_moves
+
+            late_moves.write({'delay_alert_date': new_date})
+            ontime_moves.write({'delay_alert_date': False})
+
+        ontime_moves = self.browse()
+        late_moves = self.browse()
+
+        # Check if `self` is scheduled before the previous moves. If so, `self` if late.
+        previous_done_moves = self.browse()
+        previous_nondone_moves = self.browse()
+        previous_moves_dates = []
+        for move in self.move_orig_ids:
+            if move.state == 'done':
+                previous_done_moves |= move
+            elif move.state != 'cancel':
+                previous_nondone_moves |= move
+        previous_moves_dates += previous_done_moves.mapped('date')
+        previous_moves_dates += previous_nondone_moves.mapped('date_expected')
+        if previous_moves_dates:
+            previous_moves_date = max(previous_moves_dates)
+            if new_date < previous_moves_date:
+                late_moves |= self
+            else:
+                ontime_moves |= self
+
+            late_moves.write({'delay_alert_date': previous_moves_date})
+            ontime_moves.write({'delay_alert_date': False})
```

Comment:
``` python3
            if new_date < previous_moves_date:
                self.write({'delay_alert_date': previous_moves_date})
            else:
                self.write({'delay_alert_date': False})
```

Reply:
@amoyaux with this, we will perform multiple write, instead of just 2 write, is it less performant ?